tests/e2e/onboarding.test.ts

```diff
@@ -131,6 +140,134 @@ test('onboarding with a short code', async ({ page, getOnboardingData }) => {
 	await expect(page).toHaveURL(`/onboarding`)
 })
 
+test('onboarding with GitHub OAuth', async ({ page }) => {
+	const code = `${MOCK_CODE_GITHUB}_${crypto.randomUUID()}_PLAYWRIGHT_TEST`
+
+	await page.route(/\/auth\/github(?!\/callback)/, async (route, request) => {
+		const headers = { ...request.headers(), [MOCK_CODE_GITHUB_HEADER]: code }
+		await route.continue({ headers })
+	})
+
+	const ghUser = await insertGitHubUser(code)!
+
+	await page.goto('/signup')
+	await page.getByRole('button', { name: /signup with github/i }).click()
+
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+	expect(
+		page.getByText(new RegExp(`welcome aboard ${ghUser.primaryEmail}`, 'i')),
+	).toBeVisible()
+
+	// fields are pre-populated for the user
+	const usernameInput = page.getByRole('textbox', { name: /username/i })
+	expect(usernameInput).toHaveValue(
+		// @ts-ignore
```

Comment:
Why is this failing for TypeScript? 👀 Let's look deeper into this and resolve it so we don't have to use `@ts-ignore`, it's a bad practice.

Reply:
I only when for `ignore` directive because we are in tests and so are safely proving that things are correct _at runtime_. But I totally see your point and will now iterate over it once again.

Reply:
Oh that was prosaic one: `toHaveValue` exprects a string or RegEx, while the return val of `normalizeUsername` is string or undefined. I did not give it enough info, like if string comes in, string definitely comes out and you should not be yelling at me :) Anyhow, I hated the fact that `normalizeUsername` can receive undefined, so I adjusted it to work with strings only.

Reply:
> like if string comes in, string definitely comes out and you should not be yelling at me

This sounds like a perfect use case for [generic types](https://kettanaito.com/blog/making-sense-of-typescript-generics)! Let's describe precisely this intention in types and get the correct type validation here without having to resort to ignoring the typecheck.

Reply:
True, but we do not resort to `ignore` any longer. Again, the `normalizeUsername` works with strings and so compiler can now see we are calling  `toLowerCase` on a string and returning the result